base/compiler/ssair/passes.jl

```diff
@@ -767,6 +767,21 @@ function perform_lifting!(compact::IncrementalCompact,
             else_result = lifted_value(compact, old_node_ssa, else_result,
                                        lifted_philikes, lifted_leaves, reverse_mapping)
 
+            # In cases where the Core.ifelse condition is statically-known, e.g., thanks
+            # to a PiNode from a guarding conditional, replace with the remaining branch.
+            if then_result === SKIP_TOKEN || else_result === SKIP_TOKEN
+                only_result = (then_result === SKIP_TOKEN) ? else_result : then_result
+
+                # Replace Core.ifelse(%cond, %a, %b) with %a
+                compact[lf.ssa][:inst] = only_result
```

Comment:
On second look, this bypasses the count removal of the cond. This probably needs to be:
```
compact[lf.ssa] = nothing
compact[lf.ssa][:inst] = only_result
```

Possibly also consider turning on the oracle check for the tests, which would have caught this.